components/new-contribution-flow/StepDetails.js

```diff
@@ -54,7 +54,7 @@ const StepDetails = ({ onChange, data, collective, tier, showFeesOnTop }) => {
                 type="number"
                 min={1}
                 step={1}
-                max={tier.maxQuantity}
+                max={tier?.maxQuantity}
```

Comment:
If tier is null, we won't enter the `tier?.type === TierTypes.TICKET` block

```suggestion
                max={tier.maxQuantity}
```